### PR TITLE
Manure decomposition and  assimilation sequence fix

### LINKED ISSUE #443: Execute decomposition, then assimilation processes, in `phosphorus_cycling/manure.py`
# Problem
The `Manure` module (that is within the Crop and Soil's phosphorus cycling submodule) treats the processes for decomposing and assimilating manure on the soil surface as simultaneous. This is a problem because when removing decomposed and assimilated manure mass/phosphorus, it does not upper bound the sum of the changes to the amount of manure/phosphorus that is on the field -> it is possible for the model to think that it decomposes and assimilates more manure/phosphorus than is physically possible.

# Solution
Calculate decomposed manure/phosphorus first, then remove those amounts from their respective pools, *then* calculate assimilated manure/phosphorus amounts and remove them.

Also, there should be an annual tracker added for the amount of manure mass that gets decomposed. It is important to track this for mass balance, and to better estimate the emissions from the field!

These changes need to be made in the method `daily_manure_update()` in `RUFAS/routines/soil/phosphorus_cycling/manure.py`.

### PR DESCRIPTION
**Pull request by matthew7838**
Fix the decomposition and assimilation sequence in `RUFAS/routines/soil/phosphorus_cycling/manure.py`.
### Context

Issue(s) closed by this pull request: closes #443 

### What

The sequence of decomposition and subtracting matters needs to be changed. The amount of mineralization also needs to be tracked.

### Why

The Manure module within the Crop and Soil's phosphorus cycling submodule treats the processes for decomposing and assimilating manure on the soil surface as simultaneous. This causes the possibility for the model to decompose and assimilate more than it should.

### How

Calculate decomposed manure/phosphorus first, then remove those amounts from their respective pools, then calculate assimilated manure/phosphorus amounts and remove them. Annual tracker call `total_decomposed_manure`  added for the amount of manure mass that gets decomposed.

### Test plan

1. Run with updated test suits to make sure matters are tracked correctly.
2. SME's look into this PR.

### CODE REVIEW

File: changelog.md
Comment (Isgotkowitz) [inline changelog.md]: This changelog entry needs to go in the "Next Version Updates" section

File: RUFAS/routines/field/soil/manure_pool.py
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/manure_pool.py]: Is this the total manure that has been decomposed throughout the entire life of this ManurePool (i.e. the simulation), or is it just the amount of manure that has been decomposed since the last application? I ask because there are some pools in ManurePool which are reset when manure gets applied.

Edit: read the description more carefully, it appears that this pool should be reset annually. How is that being accomplished in this PR? Also, the variable name and docstring should mention that this is an annual tracker.

**Review by Isgotkowitz — COMMENTED** [review]
This is a good start towards making the manure decomp code simpler! However I noticed a few things that I think need to be addressed.

**Review by adhikaripradip75 — COMMENTED** [review]
This section (assimilation portion) looks ok to me, however, I have several questions related to P runoff factors, that I did not see in the codes. Factors that are most important for P runoff are field slope%,, slope length, and management factor. These will be field-specific and can be included as input in the soil file. Not sure these are included but I am not able to visualize it. 

Also we are getting dry_mass of manure to the field  and again using moisture factor. Kind of contradictory. It should be mass of manure only to the field.  Again the moisture fraction in line no 19 of manure_pool.py is 0, and 0.9 and in line no 411 it is 0 and 1. The moisture fraction should be 0 and 1. 

In addition, I see the word **leached** eg in line 537 and in several places of manure_pool.py, meaning leaching nutrients to the groundwater table. The correct term  will be phosphorus pool that is being **released**  from . I will be very to discuss how to about this issue. Plase let me know.

**Review by adhikaripradip75 — APPROVED** [review]

File: changelog.md
Comment (Isgotkowitz) [inline changelog.md]: I appreciate you removing this from the wrong spot, but can you please replicate it in the "Next Version Updates" section? It is important that we do not lose track of the changes

File: changelog.md
Comment (Isgotkowitz) [inline changelog.md:37]: I think this needs to be added back in

File: RUFAS/routines/field/soil/soil_data.py
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil_data.py:423]: The annual decomposed manure should be reported through the `FieldDataReporter` in `send_annual_variables`. Otherwise users will never be able to access this data.

**Review by Isgotkowitz — COMMENTED** [review]
I think the code looks good, but there are a couple things that need to be addressed before approval.
1. The changelog needs to be corrected. There are some entries from other PRs being removed, and this is something that really needs to be avoided.
2. The annual decomposition pool needs to be reported to OutputManager, otherwise it will not be of any use to users.
3. There should be a more flushed out plan for the testing strategy. Having @adhikaripradip75 review this PR is good, but what exactly is being reviewed? Are the results from different simulations different from the results of the same simulation on `dev`? If they are not the same, is the difference tolerable?

File: RUFAS/routines/field/soil/soil_data.py
Comment (matthew7838) [reply RUFAS/routines/field/soil/soil_data.py:423]: Thanks for catching this.

**Comment by Isgotkowitz** [comment]
> This section (assimilation portion) looks ok to me, however, I have several questions related to P runoff factors, that I did not see in the codes. Factors that are most important for P runoff are field slope%,, slope length, and management factor. These will be field-specific and can be included as input in the soil file. Not sure these are included but I am not able to visualize it.

These things do factor into the P phosphorus, but they are handled in the erosion model, not the P manure model.
 
> Also we are getting dry_mass of manure to the field and again using moisture factor. Kind of contradictory. It should be mass of manure only to the field. Again the moisture fraction in line no 19 of manure_pool.py is 0, and 0.9 and in line no 411 it is 0 and 1. The moisture fraction should be 0 and 1.

The amount of manure is tracked on a dry matter basis, but we still want to know how much moisture is in it. I'm not sure why there is the difference in range between moisture factors is, but you may need to consult Pete Vadas on why that is.

> In addition, I see the word **leached** eg in line 537 and in several places of manure_pool.py, meaning leaching nutrients to the groundwater table. The correct term will be phosphorus pool that is being **released** from . I will be very to discuss how to about this issue. Plase let me know.

It is good to know the difference in terminology! Some of the P released from the manure will go into the soil, and will eventually reach the water table. But some will runoff the field.

File: changelog.md
Comment (Isgotkowitz) [inline changelog.md]:
```suggestion
```
These are duplicates

**Review by Isgotkowitz — APPROVED** [review]
Nice

**Pull request merged**
